### ISSUE TITLE
Getting a "SAML Response is not yet valid." after configuring SAML with Authentik

### ISSUE THREAD

**Issue by Dan6erbond** [gristlabs/grist-core#404]
Hello! I was following the guide to configure SAML with Authentik, and after figuring out that I needed to set `GRIST_SAML_IDP_UNENCRYPTED` to `1`, Authentik's SSO _almost_ works. I get this message in Grist, which tells me that there's some skew between the two services, maybe because of timezones, but I have no idea how to fix it.

I see this in Grist after signing-in:

![image](https://user-images.githubusercontent.com/32235074/213034112-60746792-9978-45af-88c7-01f84e2709e4.png)

It looks like Grist's default config is 5s for time skew, which is reasonable, but it cannot be changed so I wasn't able to test different values. I also have the timezone explicitly set in both containers for Authentik and Grist with `TZ` = `Europe/Zurich` in hopes that they both sync up. I appreciate if someone can help me out with this!

**Comment by paulfitz** [comment]
Authentik may not like timezones: https://github.com/goauthentik/authentik/issues/3005 - have you tried keeping it to UTC?

**Comment by paulfitz** [comment]
(crossposted at https://community.getgrist.com/t/getting-a-saml-response-is-not-yet-valid-error-in-grist-with-authentik/1977)

**Comment by Dan6erbond** [comment]
Hey! Thanks for the super quick response. I tried setting the timezones in both containers to UTC, but get the same error. Is there anything I can try to test or validate that this change is making a difference?

**Comment by paulfitz** [comment]
Hmm perhaps you could `docker exec -it <container-name> /bin/bash` into each container and poke around a bit, check the output of `date`  looks good and with seconds in sync with wall time for both? I don't think there's any useful logging about this on the Grist side, maybe there is something on the Authentik side. And have you tried not setting the timezone at all for either?

**Comment by Dan6erbond** [comment]
Hey, sure. I'm using K3s actually to deploy both apps. Here are the outputs of `date` in both containers. Left is Grist, right is Authentik's server. But all the Authentik components (server, worker, LDAP outpost, etc.) share the same environment:

![image](https://user-images.githubusercontent.com/32235074/213403804-5a15b15d-c4b8-4eb6-9d58-a8961fba4642.png)

Initially I didn't have the timezone set in either container. I only tried that after realizing that the error was related to desync between the servers.

**Comment by Dan6erbond** [comment]
I checked Authentik's logs as well, only see the authorization event with the SAML Request. Any way I can parse that? But everything looks fine, since Authentik is authorizing me and fowarding to Grist.

**Comment by Dan6erbond** [comment]
Oh man, I just figured it out! I went through all the steps, increasing skew time in Grist's source files, built a Docker container, spun up the container with Docker Compose and created a new Provider in the Authentik UI for the test instance.

I was logged-in and had some other issues locally, so I decided to compare the Authentik providers between the one I created manually for local debugging and the one I deployed with Terraform - Terraform's provider automatically sets "Assertion not valid before" incorrectly to `minutes=3`. Once I changed it to `minutes=-3` it worked!

Only thing now is that I'm getting "Access denied" to the team workspace. But it might be because I initially didn't have my email set in the `GRIST_DEFAULT_EMAIL` value. I'll redeploy Grist and see if it works now. Thanks for all your help and quick responses!
